windows.js

```diff
@@ -50,15 +54,70 @@ export async function install(platform, engine, version) {
 
   let toolchainPaths = (version === 'mswin') ? await setupMSWin() : await setupMingw(version)
 
-  common.setupPath([`${rubyPrefix}\\bin`, ...toolchainPaths])
-
   if (!inToolCache) {
     await downloadAndExtract(engine, version, url, base, rubyPrefix);
   }
 
+  const winMSYS2Type = common.setupPath([`${rubyPrefix}\\bin`, ...toolchainPaths])
+
+  // install msys2 tools for all Ruby versions, only install mingw or ucrt for Rubies >= 2.4
+
+  if (!['windows-2019', 'windows-2016'].includes(virtualEnv)) {
```

Comment:
A path check here would be better as well, and we can have a comment this is the case on windows-2022

Reply:
One is being used to check whether to update/install `C:/msys64/usr` files, and one is used for `C:/msys64/mingw64` or `C:/msys64/ucrt64` files.  Not sure, but added a constant...

Reply:
One is being used to check whether to update/install `C:/msys64/usr` files, and one is used for `C:/msys64/mingw64` or `C:/msys64/ucrt64` files.  Not sure, but added a constant...

Reply:
Is there an easy way to check if some path/dir/file exist to asses if MSYS2 is preinstalled? I would prefer to avoid depending on a given image having MSYS2 installed or not if possible.

Reply:
I did think about this.  The base MSYS2 installation (bash shell & pacman) is installed on windows-2022.  What is not installed are the 'shared build tools' that are not dependent on the build toolset.  It's very similar to the bash shell that git installs, with the addition of pacman.

From the discussions in actions/virtual-environments, I've gotten the impression that they will lock what MSYS2 packages are installed on images.  IOW, if there are changes, they will not add packages to a pre-existing image, they will only do so when a new image is released.  Also, I doubt they will make changes to the MSYS2 install on new images in the future.

Reply:
OK. I think it could be an issue for self-hosted windows, but I guess that's very rare and not supported currently, so fine as-is.